src/Tasks/ComReference.cs

```diff
@@ -407,21 +407,28 @@ internal static string StripTypeLibNumberFromPath(string typeLibPath, FileExists
         private static string GetModuleFileName(IntPtr handle)
         {
             bool success = false;
-            var buffer = new StringBuilder();
+            char[] buffer = null;
+            int pathLength = 0;
 
             // Try increased buffer sizes if on longpath-enabled Windows
             for (int bufferSize = NativeMethodsShared.MAX_PATH; !success && bufferSize <= NativeMethodsShared.MaxPath; bufferSize *= 2)
             {
-                buffer.EnsureCapacity(bufferSize);
-
-                var handleRef = new System.Runtime.InteropServices.HandleRef(buffer, handle);
-                int pathLength = NativeMethodsShared.GetModuleFileName(handleRef, buffer, buffer.Capacity);
+                buffer = System.Buffers.ArrayPool<char>.Shared.Rent(bufferSize);
```

Comment:
should I use stackalloc here?

Reply:
GetModuleFileName can have a max buffer size of 260. So I think using `ArrayPool` is probably appropriate.